node/lib/DeviceAccessor.ts

```diff
@@ -1,18 +1,89 @@
 
 import { DeviceInterface } from "./DeviceInterface";
-import { IDevice } from "./ITranslator";
+import { ITranslator } from "./ITranslator";
 
 import { EventEmitter } from "events";
 
 /**
  * Provides reflection-style access to device properties and methods via device interfaces.
  */
 export class DeviceAccessor {
+    /**
+     * Loads a DeviceInterface from a module. This is just a convenience wrapper
+     * around require(). Throws if the module could not be loaded.
+     *
+     * @param {string} interfaceModuleName  Package-qualified name (or relative path) of
+     *     the interface module. The package-qualified name can be obtained from
+     *     PackageTranslatorInfo.interfaces.
+     *
+     * @returns {Promise<DeviceInterface>} The loaded device interface.
+     */
+    public static getInterfaceAsync(interfaceModuleName: string): Promise<DeviceInterface>;
```

Comment:
No, it won't need to change. This only gets a single interface. If you want all the interfaces that a translator implements, then you need to call this once for every item in the `PackageTranslatorInfo.interfaces` array. I considered adding a method that takes and returns an array, but it didn't seem necessary.

Additionally in the future interfaces may themselves reference other interfaces: "composability at the schema level" I guess. That is covered by the `DeviceInterface.references` property.

---
In reply to: [73273050](https://github.com/openT2T/opent2t/pull/2#discussion_r73273050) [](ancestors = 73273050)